app/controllers/users_controller.rb

```diff
@@ -1,24 +1,19 @@
 class UsersController < ApplicationController
-  before_action :set_user, only: [:show, :destroy]
-  before_action :verify_ownership, only: [:destroy]
+  before_action :downcase, only: [:show]
+  before_action :set_user, only: [:show]
 
   def show
   end
 
-  def destroy
-    current_user.runs.destroy_all if params[:destroy_runs] == '1'
-    current_user.destroy
-    auth_session.invalidate!
-    redirect_to root_path, alert: 'Your account has been deleted. Go have fun outside :)'
-  end
-
   private
 
   def set_user
     @user = User.find_by(name: params[:user]) || not_found
   end
 
-  def verify_ownership
-    render :unauthorized, status: 401 unless @user == current_user
+  def downcase
+    if params[:user] != params[:user].downcase
+      redirect_to user_path(params[:user].downcase)
```

Comment:
Is there a reason for the hard redirect here?

Reply:
Yeah, I meant leaving it as is and just doing the lookup with the downcased username.  That was you don't add a full round trip to any request that doesn't lowercase the username (and I figured there would be a lot of that since you explicitly state you can link it in any casing on one of the pages).

Reply:
I like the behavior of forcing lowercase mostly for philosophical reasons about URLs. I believe in one-URL-per-page for the internet in general.

There are some minor benefits such as helping the belief that changing your username case doesn't matter (because you already have seen URLs in full-lower that access users whose names aren't).

I've also experienced hidden caching issues in another project after not realizing we were double- or triple- (or more) caching pages because multiple URLs rendered the same user profile page. I don't think we have anything like that in Splits I/O right now, but it's just one low-chance possible future bug bite.

For supporting arguments, see the [W3 spec on HTML and URLs][1], which states "users should always consider that URLs are case-sensitive" combined with the [robustness principle][2]'s "be conservative in what you send, be liberal in what you accept" (the at-rest URL being something we "send" since people copy-paste it).

[1]: https://www.w3.org/TR/WD-html40-970708/htmlweb.html
[2]: https://en.wikipedia.org/wiki/Robustness_principle

Reply:
Very good reasons, glad I asked because I didn't know the spec said to treat them as case sensitive.  Learn something new everyday.